astropy/io/fits/header.py

```diff
@@ -1694,7 +1694,7 @@ def _keyword_from_index(self, idx):
         """
 
         if idx < 0:
-            idx += len(self._cards) - 1
+            idx += len(self._cards)
```

Comment:
Should we give @mohanagr some credit here by separating this out as its own commit and make him the author of that commit? I prefer this PR over #5852 but he did get the ball rolling, and his PR will never be merged if we accept this one instead. Just a thought.